### PR TITLE
Added support for kerberos

### PR DESCRIPTION
**Pull request by bbaugher**
Test evidence: https://gist.github.com/bbaugher/89fb03c794e77c8d62fd822789bfa7b4

### CODE REVIEW

File: spec/default_spec.rb [hunk 1]
```diff
@@ -35,4 +35,54 @@
...
+          node.override["confluent"]["kerberos"]["keytab"] = '/path/to/keytab'
+        end
+      end
+
+      it 'should raise exception' do
+        expect { chef_run.converge(described_recipe) }.to raise_error(RuntimeError)
+      end
+    end
+
+    context 'and keytab_location not specified' do
```
Comment (noslowerdna) [inline spec/default_spec.rb]: keytab_location -> keytab location

File: templates/default/jaas.conf.erb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+KafkaServer {
```
Comment (noslowerdna) [inline templates/default/jaas.conf.erb:2]: add the standard auto-generated by Chef comment

File: spec/default_spec.rb (hunk 1 above)
Comment (bbaugher) [reply spec/default_spec.rb]: Done, https://github.com/bbaugher/confluent/pull/18/commits/d3d8aae3eb27f12cbac21f5fef67c6e7df9cef13

File: templates/default/jaas.conf.erb (hunk 2 above)
Comment (bbaugher) [reply templates/default/jaas.conf.erb:2]: Done, https://github.com/bbaugher/confluent/pull/18/commits/73e9117d5f9f677cf2eac9fece01c6f9b0de4397

**Comment by noslowerdna** [comment]
+1

File: .rubocop.yml [hunk 3]
```diff
@@ -215,3 +215,6 @@ Style/TrailingWhitespace:
 # Configuration parameters: EnforcedStyle, SupportedStyles.
 Style/VariableName:
   Enabled: false
+
+Style/FrozenStringLiteralComment:
+  Enabled: false
```
Comment (cduranleau) [inline .rubocop.yml]: Why disable vs. follow the pattern?

File: attributes/default.rb [hunk 4]
```diff
@@ -106,3 +106,15 @@
 default["confluent"]["zookeeper"]["zookeeper.properties"]["dataDir"] = "/tmp/zookeeper"
 default["confluent"]["zookeeper"]["zookeeper.properties"]["clientPort"] = "2181"
 default["confluent"]["zookeeper"]["zookeeper.properties"]["maxClientCnxns"] = "0"
+
+# Kerberos configuration
+default["confluent"]["kerberos"]["enable"] = false
```
Comment (cduranleau) [inline attributes/default.rb]: Nit: needs more single quotes vs. double quotes.

File: recipes/_install.rb [hunk 5]
```diff
@@ -78,3 +78,17 @@
 link "/etc/schema-registry" do
   to "#{node["confluent"]["install_dir"]}/confluent-#{node["confluent"]["version"]}/etc/schema-registry"
 end
+
+# Write JAAS configuration file if enabled
+if node["confluent"]["kerberos"]["enable"]
+  # Verify required attributes are set
+  raise "Kerberos keytab location must be configured" if node["confluent"]["kerberos"]["keytab"].nil?
+  raise "Kerberos realm or principal must be configured" if node["confluent"]["kerberos"]["principal"].end_with? '@'
```
Comment (cduranleau) [inline recipes/_install.rb]: Unrelated question, but is it possible for a chef-node to NOT have a FQDN?

Comment (bbaugher) [reply recipes/_install.rb]: Not that I'm aware of nor have I ever seen that happen. Its an ohai attribute so its not even settable

**Comment by cduranleau** [comment]
+1

File: .rubocop.yml (hunk 3 above)
Comment (bbaugher) [reply .rubocop.yml]: Removed and fixed, https://github.com/bbaugher/confluent/pull/18/commits/2bde1faa3482fbfb3fd9c88caa1a327bad058cf0

File: attributes/default.rb (hunk 4 above)
Comment (bbaugher) [reply attributes/default.rb]: Fixed, https://github.com/bbaugher/confluent/pull/18/commits/aae6a901395b5adacdda58fc23bbca6c20e86555

**Comment by bbaugher** [comment]
Have enough +1s, merging

**Pull request merged**